### DISCUSSION TITLE
第六章中，为什么用一个结构体来代替（address，uint256）？

### DISCUSSION THREAD

**Discussion by kugool-sy** [smartcontractkit/Web3_tutorial_Chinese#17] [Q&A]
第六章中，为什么encode消息时使用的是uint256和address，而decode时却用一个结构体RequestData来接收数据，不应该使用uint256和address来接收吗

**Comment by QingyangKong** [comment]
你是指哪个合约？你能把相关代码写出来吗？

**Reply by kugool-sy** [reply]
NFTPoolLockBurnAndMint.sol中引入了RequestData结构体用来接收Decode数据。数据原来是(uint256,address) 进行encode得到的，虽然用这个结构体来接收是可以正确解析的，但这里有什么考虑呢？为什么不直接用（uint256,address）来直接接收数据？
